classes/class-p4-context-controller.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * P4 Context Controller Class
+ *
+ * @package P4MT
+ */
+
+/**
+ * Class P4_Context_Controller Sets common context fields.
+ */
+class P4_Context_Controller {
```

Comment:
You can leave out the `_Controller` suffix here. A lot of things are called a "controller" so it doesn't give you any more information than just `P4_Context` :)